runtime/compiler/optimizer/InterpreterEmulator.cpp

```diff
@@ -692,10 +1220,21 @@ InterpreterEmulator::updateKnotAndCreateCallSiteUsingInvokeCacheArray(TR_Resolve
 bool
 InterpreterEmulator::isCurrentCallUnresolvedOrCold(TR_ResolvedMethod *resolvedMethod, bool isUnresolvedInCP)
    {
+   if (!resolvedMethod)
+      return true;
+
+   // Coldness info in thunk archetype is not accurate
+   // If we can get the resolved method, then it shouldn't be cold
+   // TODO: customized lambda form method is similar, it may not
+   // be executed enough before it gets inlined
+   //
+   if (_callerIsThunkArchetype)
```

Comment:
@vijaysun-omr Another problem with previous commit is that it refactored the code such that we only compute return value of non-cold calls. However, because thunk archetypes are never interpreted, so the call bytecodes always appear as cold. This includes the call to `MutableCallSite.getTarget()`, whose return value is used to guide MutableCallSite inlining. MutableCallSite inlining is broken because previous commit doesn't compute this call's value due to incorrect coldness info.

Reply:
Because `MutableCallSite.getTarget()` is treated as cold in the baseline, it won't be inlined in the first inlining pass (will be inlined in targeted inlining only if a pass is requested), so the fix here will inline `MutableCallSite.getTarget()` in the first pass which gives us performance gain in jruby benchmarks.

Reply:
Okay, this sounds critical and I doubt that filtering on cold paths in this code is the right way to go.

Reply:
One problem is the check on `isUnresolvedInCP`. Since the thunk archetypes are never interpreted, and we always do compile time resolve on them, so the majority of its cp entries will appear unresolved. I think we should at least ignore `isUnresolvedInCP` for calls in thunk archetypes. @vijaysun-omr What do you think?

Reply:
Yes that makes sense. Perhaps these routines from the cold block marker code help you with these sorts of cases :

https://github.com/eclipse/omr/blob/cf3ef5ce33a7993f5749463a17d39e4c8e6b6db0/compiler/optimizer/LocalOpts.cpp#L7924

and 

https://github.com/eclipse/omr/blob/cf3ef5ce33a7993f5749463a17d39e4c8e6b6db0/compiler/optimizer/LocalOpts.cpp#L7986